### PR TITLE
drivers: i2c: Add Xilinx PS i2c driver

### PR DESCRIPTION
**Pull request by gitGideonhub**
Add Xilinx PS i2c driver for Ultrascale+ and this driver ONLY tested on MPSoC(UG1085/embeddedsw). Besides, this driver remaining handle the send/read data larger than fifo depth(14 Bytes).

Note: I think the naming of Kconfig.xilinx_axi can be changed to Kconfig.xilinx and also the "DT_HAS_XLNX_XPS_IIC_ENABLED" dts config need to be changed too.

### CODE REVIEW

**Comment by teburd** [comment]
Please fix the CI issues

**Comment by gitGideonhub** [comment]
Hi @teburd, thanks for getting back to me.
Closely looking this CI fail which is happen on dependency of device tree config in i2c driver config and the "DT_HAS_XLNX_XPS_IIC_ENABLED" config label is auto generate by west tool chain (I think).

The question I want to ask is should I remove the depend in i2c driver config or somehow add the zephjr/build/Kconfig(auto gen file) in it

**Comment by gitGideonhub** [comment]
Hi, @teburd 
Sorry for lately update about the yaml file. please check it after CI is done

**Review by teburd — APPROVED** [review]

File: dts/bindings/i2c/xlnx,xps-iic.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+description: Xilinx PS IIC Bus Interface
+
+compatible: "xlnx,xps-iic"
```
Comment (henrikbrixandersen) [inline dts/bindings/i2c/xlnx,xps-iic.yaml:4]: The I2C IPs present in the AMD Xilinx MPSoC are from Cadence, not from Xilinx.
This should be a Cadence driver matching `compatible = "cdns,i2c-r1p14"`

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Comment by neeliajay** [comment]
Hi @gitGideonhub 
I noticed that this PR has been open for a while.
Are you still planning to continue working on it?

If you're unable to proceed, I'm happy to take over and help implement any necessary updates or changes.
Let me know how you'd like to proceed!

**Comment by gitGideonhub** [comment]
> Hi @gitGideonhub I noticed that this PR has been open for a while. Are you still planning to continue working on it?
> 
> If you're unable to proceed, I'm happy to take over and help implement any necessary updates or changes. Let me know how you'd like to proceed!

Hello @neeliajay,
I am so glad to know this info, please modify the vernor please!, I am too busy on work to maintain this PR".
Appreciating for  your support!

**Pull request closed without merging**